hosts/groups/meta/hardware.nix

```diff
@@ -0,0 +1,43 @@
+{
```

Comment:
I don't think this belongs in `/hosts`

Take a look at `/nixosModules/providers`, where we have a common config for e.g. `equinix`.
Similarly, we could add `/nixosModules/providers/profian` or `./nixosModules/hardware/nuc` etc.

Also note that https://github.com/NixOS/nixos-hardware exists, which we can probably reuse here for real hardware

A big benefit of having this in `nixosModules` is that then you can e.g. refer to it in an external project via `github:profianinc/infrastructure#nixosModules.hardware.nuc` (or whatever else), which is extremely useful for e.g. bootstraping.

Reply:
This module is not intended for "all hardware quirks" or anything like that, but just for setting up some things that are common for bare-metal deployed machines. Like where it can find its root and swap partitions.
I don't see why we'd need to be able to import this from an external project, if it's just in order to set parameters for our machines?